File: deploy/dux-front/lib/features/timetree/domain/models/timetree_notification.dart
import 'package:dux_front/features/timetree/data/dto/timetree_notification_dto.dart';

class TimetreeNotification {
  final String id;
  final String recipientId;
  final String title;
  final String content;
  final String type;
  final String entityType;
  final String entityId;
  final String actionType;
  final bool isRead;
  final DateTime createdAt;

  const TimetreeNotification({
    required this.id,
    required this.recipientId,
    required this.title,
    required this.content,
    required this.type,
    required this.entityType,
    required this.entityId,
    required this.actionType,
    required this.isRead,
    required this.createdAt,
  });

  factory TimetreeNotification.fromDto(TimetreeNotificationDto dto) {
    return TimetreeNotification(
      id: dto.id,
      recipientId: dto.recipientId,
      title: dto.title,
      content: dto.content,
      type: dto.type,
      entityType: dto.entityType,
      entityId: dto.entityId,
      actionType: dto.actionType,
      isRead: dto.isRead,
      createdAt: dto.createdAt,
    );
  }

  TimetreeNotificationDto toDto() {
    return TimetreeNotificationDto(
      id: id,
      recipientId: recipientId,
      title: title,
      content: content,
      type: type,
      entityType: entityType,
      entityId: entityId,
      actionType: actionType,
      isRead: isRead,
      createdAt: createdAt,
    );
  }
}

File: deploy/dux-integration/src/main/java/com/asm/dux/timetree/domain/TimetreeNotification.java
package com.asm.dux.timetree.domain;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "TT_NOTIFICATION", schema = "dbo")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TimetreeNotification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID")
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "RECIPIENT_ID", nullable = false)
    private Member recipient;

    @Column(name = "TITLE", nullable = false, length = 150)
    private String title;

    @Column(name = "CONTENT", nullable = false, length = 1000)
    private String content;

    @Column(name = "TYPE", nullable = false, length = 50)
    private String type; // e.g. NEW_MESSAGE, NEW_ATTACHMENT, etc.

    @Column(name = "ENTITY_TYPE", nullable = false, length = 50)
    private String entityType; // e.g. EVENT, ATTACHMENT, MESSAGE

    @Column(name = "ENTITY_ID", nullable = false)
    private Long entityId;

    @Column(name = "ACTION_TYPE", nullable = false, length = 50)
    private String actionType; // e.g. CREATED, UPDATED, DELETED, NEW, ASSIGNED

    @Column(name = "IS_READ", nullable = false)
    @Builder.Default
    private boolean read = false;

    @Column(name = "CREATED_AT", nullable = false)
    private LocalDateTime createdAt;
}
